### PR TITLE
android-tv: player add video player events

### PR DESCRIPTION
**Pull request by shaymark**
adding events to the player
in order to support 
https://applicaster.atlassian.net/browse/IM-534

sending this events:
onVideoProgress
onVideoStart
onVideoPause
onVideoSeek
onVideoEnd
onVideoError

### CODE REVIEW

File: tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt [hunk 1]
```diff
@@ -88,29 +112,45 @@ class TVAQuickBrickPlayerView(context: Context, attrs: AttributeSet?) : FrameLay
...
             analyticUtil.trackPause(it.time, bitmovinPlayer?.duration ?: 0.0)
+            quickBrickPlayer.onPause()
         })
         addEventListener(OnPlaybackFinishedListener {
             trackTime(true)
+            quickBrickPlayer.onEnd()
         })
         addEventListener(OnSeekListener {
             analyticUtil.trackSeek(it.position, it.seekTarget, bitmovinPlayer?.duration ?: 0.0)
+            quickBrickPlayer.onSeek(it.position * 1000L, it.seekTarget.toLong() * 1000L)
```
Comment (PRuedaApplicaster) [inline tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt:135]: Don't hardcode number please.

File: tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt [hunk 2]
```diff
@@ -88,29 +112,45 @@ class TVAQuickBrickPlayerView(context: Context, attrs: AttributeSet?) : FrameLay
...
+        })
+        addEventListener(OnReadyListener {
+            playbackProgressObservable =
+                    Observable.interval(1, TimeUnit.SECONDS)
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe { p ->
+                        quickBrickPlayer.onTimeUpdate(
+                                (bitmovinPlayer?.currentTime?.toLong() ?: 0L * 1000),
+                                (bitmovinPlayer?.duration?.toLong() ?: 0L * 1000) ?: 0L
```
Comment (PRuedaApplicaster) [inline tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt:144]: Don't hardcode number please.

File: tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt [hunk 3]
```diff
@@ -177,10 +209,11 @@ class TVAQuickBrickPlayerView(context: Context, attrs: AttributeSet?) : FrameLay
     @RequiresApi(Build.VERSION_CODES.O)
     private fun audioFocusRequest() =
             AudioFocusRequest.Builder(AudioManager.AUDIOFOCUS_GAIN)
+                    .setOnAudioFocusChangeListener(audioFocusListener)
                     .setAudioAttributes(
                             AudioAttributes.Builder()
                                     .setUsage(AudioAttributes.USAGE_MEDIA)
-                                    .setContentType(AudioAttributes.CONTENT_TYPE_MOVIE)
+                                    .setContentType(AudioAttributes.CONTENT_TYPE_MUSIC)
```
Comment (PRuedaApplicaster) [inline tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt:216]: Why do we need to change this?

File: tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt [hunk 4]
```diff
@@ -388,4 +420,17 @@ class TVAQuickBrickPlayerView(context: Context, attrs: AttributeSet?) : FrameLay
...
+    private fun timeUpdate(): Observable<Long> {
+        return Observable.interval(1, TimeUnit.SECONDS)
+                .map { checkIfPlayerIsPlaying() }
+    }
+
+    private fun checkIfPlayerIsPlaying(): Long {
+        if (bitmovinPlayer?.isPlaying!!) {
+            return bitmovinPlayer?.currentTime?.toLong() ?: 1
+        }
+        return 1.toLong()
```
Comment (PRuedaApplicaster) [inline tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt:433]: Please, no hardcode numbers.

File: tvos/plugin/src/android/index.js [hunk 5]
```diff
@@ -54,7 +54,7 @@ export class AndroidPlayer extends React.Component<Props, State> {
     DeviceEventEmitter.addListener("emitterOnError", this.onError);
     DeviceEventEmitter.addListener("onTvKeyDown", this.onKeyDown);
     DeviceEventEmitter.addListener("onShowSettings", this.onShowSettings);
-    sendQuickBrickEvent("blockTVKeyEmit", { blockTVKeyEmit: false });
+    sendQuickBrickEvent("blockTVKeyEmit", { blockTVKeyEmit: true });
```
Comment (PRuedaApplicaster) [inline tvos/plugin/src/android/index.js:57]: Why do we need to change this to true?

Comment (shaymark) [reply tvos/plugin/src/android/index.js:57]: this i part of marge from 0.6.5 to the master, it wasn't marge

File: tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt (hunk 3 above)
Comment (shaymark) [reply tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt:216]: this i part of marge from 0.6.5 to the master, it wasn't marge

File: tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt (hunk 2 above)
Comment (shaymark) [reply tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt]: changed to constans

File: tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt (hunk 1 above)
Comment (shaymark) [reply tvos/plugin/android/src/main/kotlin/com/tva/quickbrickplayerplugin/TVAQuickBrickPlayerView.kt]: change to constants

**Pull request merged**